### PR TITLE
Refactored the listing photos and profile photos to be public

### PR DESCRIPTION
**Pull request by akashd50**
Refactored the listing photos and profile photos to be public
Refactored all the related tests
Changed the schema to store profile, banner photo and listing photo urls
Changes to the listing controllers:
Get listing also returns the listing photo urls
Delete listing also removes all the listing photos

Sorry this again is quite long but its mostly lot of files with small changes!

### CODE REVIEW

**Comment by gilhrpenner** [comment]
I think we're just missing to update the README.md with the changes on the bucket

**Review by gilhrpenner — CHANGES_REQUESTED** [review]
When uploading a profile picture the old one is not being removed from S3, not that it makes any difference for this project. It might be easier to just keep all profile pictures with the userId, wouldn't it?

**Comment by akashd50** [comment]
> 

So right now, it does remove the existing photo when we put a new profile photo, because I thought that they be essentially untracked since we only store one url at a time!

File: frontend/src/components/AccountGeneralPanel.vue [hunk 1]
```diff
@@ -176,11 +176,14 @@ const onUpdatePicture = (file: File | null) => {
...
-            store.commit('account/updateProfilePicture', newPicture);
-        }
+    type AxiosSuccessResponse = {
+        data: {
+            photoURL: string;
+        };
+    };
+
+    api.put('/photos/profile', formData).then((response: AxiosSuccessResponse) => {
+        store.commit('account/updateProfilePicture', response.data.photoURL);
```
Comment (akashd50) [inline frontend/src/components/AccountGeneralPanel.vue:186]: Hmm.. I don't think this is right, I think the correct would be `response.data.photos[0].photoUrl`. Its the genric object for get photos requests for both listings and profile photo.

Comment (akashd50) [reply frontend/src/components/AccountGeneralPanel.vue:186]: I can make this change and test it!

Comment (gilhrpenner) [reply frontend/src/components/AccountGeneralPanel.vue:186]: I did this and it worked?

**Comment by gilhrpenner** [comment]
> > 
> 
> 
> 
> So right now, it does remove the existing photo when we put a new profile photo, because I thought that they be essentially untracked since we only store one url at a time!

Can you test it locally? I tried your branch and I have multiple profile pictures there

**Comment by akashd50** [comment]
> > > 
> > 
> > 
> > So right now, it does remove the existing photo when we put a new profile photo, because I thought that they be essentially untracked since we only store one url at a time!
> 
> Can you test it locally? I tried your branch and I have multiple profile pictures there

For sure, I'm pretty sure it worked, but I'll check again!

**Comment by gilhrpenner** [comment]
> > > > 
> 
> > > 
> 
> > > 
> 
> > > So right now, it does remove the existing photo when we put a new profile photo, because I thought that they be essentially untracked since we only store one url at a time!
> 
> > 
> 
> > Can you test it locally? I tried your branch and I have multiple profile pictures there
> 
> 
> 
> For sure, I'm pretty sure it worked, but I'll check again!
> 
> 

Alright, let me know how the goes. Heading to bed now

File: frontend/src/components/AccountGeneralPanel.vue (hunk 1 above)
Comment (akashd50) [reply frontend/src/components/AccountGeneralPanel.vue:186]: Oh wait yeah, I was looking at the wrong endpoint, I think this return is inconsistent from the others, I'll change the return type and update this as well!

**Comment by akashd50** [comment]
Okay so fixed up some last bit of changes, updated readme and tested the profile photo and it does seem to work as intended. Would you try again and let me know if you still have issues @gilhrpenner!

**Review by gilhrpenner — APPROVED** [review]

**Pull request merged**